src/mpcCoreKit.ts

```diff
@@ -907,14 +923,22 @@ export class Web3AuthMPCCoreKit implements ICoreKit {
       if ((await this.checkIfFactorKeyValid(hashedFactorKey)) && !this.options.disableHashedFactorKey) {
         // Initialize tkey with existing hashed share if available.
         const factorKeyMetadata: ShareStore = await this.getFactorKeyMetadata(hashedFactorKey);
-        await this.tKey.inputShareStoreSafe(factorKeyMetadata, true);
-        await this.tKey.reconstructKey();
-        await this.finalizeTkey(hashedFactorKey);
+        try {
+          await this.tKey.inputShareStoreSafe(factorKeyMetadata, true);
+          await this.tKey.reconstructKey();
+          await this.finalizeTkey(hashedFactorKey);
+        } catch (err) {
+          log.error("error initializing tkey with hashed share", err);
+        }
       }
     }
   }
 
   private async finalizeTkey(factorKey: BN) {
+    if (this.state.accountIndex !== 0) {
+      log.warn("AccountIndex should be 0");
+      this.state.accountIndex = 0;
```

Comment:
so everytime, user does a tkey operation.. account index will reset to 0? Let say user changes account settings while using account index 1, it will reset the account to index 0. Is there any way to reset back to account index being used?

Reply:
We can use localStorage to save the state, but I prefer let the developer to handle it.
setting it to 0 on initialise keep consistent behaviour and reduce future maintenance

Reply:
We need to add it in our example usage thn in case we want developer to handle it.